internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -25,38 +27,116 @@ const (
 
 const (
 	annotationCPULoadBalancing = "cpu-load-balancing.crio.io"
+	annotationCPUQuota         = "cpu-quota.crio.io"
+	annotationIRQLoadBalancing = "irq-load-balancing.crio.io"
+	annotationTrue             = "true"
 	schedDomainDir             = "/proc/sys/kernel/sched_domain"
+	irqSmpAffinityProcFile     = "/proc/irq/default_smp_affinity"
+	cgroupMountPoint           = "/sys/fs/cgroup"
 )
 
 // HighPerformanceHooks used to run additional hooks that will configure a system for the latency sensitive workloads
 type HighPerformanceHooks struct{}
 
 func (h *HighPerformanceHooks) PreStart(ctx context.Context, c *oci.Container, s *sandbox.Sandbox) error {
 	log.Infof(ctx, "Run %q runtime handler pre-start hook for the container %q", HighPerformance, c.ID())
+
+	if isCgroupParentBurstable(s) {
+		log.Infof(ctx, "Container %q is a burstable pod. Skip PreStart.", c.ID())
+		return nil
+	}
+	if isCgroupParentBestEffort(s) {
+		log.Infof(ctx, "Container %q is a besteffort pod. Skip PreStart.", c.ID())
+		return nil
+	}
+	if !isContainerRequestWholeCPU(c) {
+		log.Infof(ctx, "Container %q requests partial cpu(s). Skip PreStart", c.ID())
+		return nil
+	}
+
 	// disable the CPU load balancing for the container CPUs
 	if shouldCPULoadBalancingBeDisabled(s.Annotations()) {
+		log.Infof(ctx, "Disable cpu load balancing for container %q", c.ID())
 		if err := setCPUSLoadBalancing(c, false, schedDomainDir); err != nil {
 			return err
```

Comment:
Let's wrap those errors to add more context:
```suggestion
			return errors.Wrap(err, "set CPU load balancing")
```

Reply:
I like it!